src/lib/builders/listbox/create.ts

```diff
@@ -366,13 +366,13 @@ export function createListbox<
 						(e.key === kbd.ENTER && !e.isComposing) ||
 						(e.key === kbd.SPACE && isHTMLButtonElement(node))
 					) {
-						e.preventDefault();
 						const $highlightedItem = highlightedItem.get();
 						if ($highlightedItem) {
+							e.preventDefault();
 							selectItem($highlightedItem);
-						}
-						if (!multiple.get()) {
-							closeMenu();
+							if (!multiple.get()) {
```

Comment:
I don't believe this should be inside the highlighted check. We should always close it when pressing enter. The prevent default looks good though!